skbeam/core/recip.py

```diff
@@ -342,3 +342,37 @@ def gisaxs(incident_beam, reflected_beam, pixel_size, detector_size,
     qr = np.sqrt(qx**2 + qy**2)
 
     return gisaxs_output(alpha_i, theta_f, alpha_f, tilt_angle, qx, qy, qz, qr)
+
+
+def generate_q_bins(rmax, qmax, pixel_size, distance, wavelength):
+    """
+    Generate the Q bins at the resolution of the detector
+    Parameters
+    -----------
+    rmax: float
+        The maximum radial distance on the detector
```

Comment:
The units don't matter so long as they are the same as the `pixel_size` and `distance`. I usually use meters, but that is only because that is what pyFAI spits out. Since we eventually take the ratio of the pixel size and the sample to detector distance we only care that they have the same units. I can include units if you want, it just might not be necessary. The function itself returns the bins in the inverse whatever units the wavelength was in.